zipline/pipeline/loaders/utils.py

```diff
@@ -116,9 +116,11 @@ def previous_event_frame(events_by_sid,
     next_date_frame
     """
     sids = list(events_by_sid)
+    populate_value = None if field_dtype == categorical_dtype else \
```

Comment:
Basically, we were using the `missing_value` of columns before in order to populate the array with default values; we can't do this with categoricals because the current design is such that they don't have a default missing value, so we have to explicitly check if the field is a categorical, and if so, here we can just use `None` as the missing value.